File: Android_Zone_Test/src/com/example/mylib_test/activity/animal/ViewHelperTestActivity.java
package com.example.mylib_test.activity.animal;

import android.view.View;
import android.widget.Button;
import android.widget.LinearLayout;

import com.example.mylib_test.R;
import com.nineoldandroids.view.ViewHelper;

import and.base.activity.BaseActivityZ;

/**
 * Created by Administrator on 2016/1/28.
 */
public class ViewHelperTestActivity extends BaseActivityZ {
    private Button bt1, bt2, bt3;
    private LinearLayout ll_main;

    @Override
    public void setContentView() {
        setContentView(R.layout.a_viewhelper);
    }

    @Override
    public void findIDs() {
        bt1 = (Button) findViewById(R.id.bt1);
        bt2 = (Button) findViewById(R.id.bt2);
        bt3 = (Button) findViewById(R.id.bt3);
        ll_main = (LinearLayout) findViewById(R.id.ll_main);
    }

    @Override
    public void initData() {

    }

    @Override
    public void setListener() {
        ll_main.setOnClickListener(this);
        bt2.setOnClickListener(this);
        bt3.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.ll_main:
                startAni();
                break;
            case R.id.bt2:
                startAni2();
                break;
            case R.id.bt3:
                startAni3();
                break;
        }
        super.onClick(v);
    }

    private void startAni() {
        ViewHelper.setY(bt1, 100);
    }

    private void startAni2() {
//        ViewHelper.setY(bt1,100);
        ViewHelper.setTranslationY(bt1, 100);
    }

    private void startAni3() {
        ViewHelper.setScrollX(ll_main, 100);
        ViewHelper.setScrollX(bt3, -100);
    }
}

File: Android_Zone_Test/src/com/example/mylib_test/activity/animal/ViewHelperTestActivity.kt
package com.example.mylib_test.activity.animal

import android.view.View

import com.example.mylib_test.R
import com.zone.lib.base.controller.activity.BaseFeatureActivity
import com.nineoldandroids.view.ViewHelper

import kotlinx.android.synthetic.main.a_viewhelper.*

/**
 * Created by Administrator on 2016/1/28.
 */
class ViewHelperTestActivity : BaseFeatureActivity() {

    override fun setContentView() {
        setContentView(R.layout.a_viewhelper)
    }

    override fun initData() {

    }

    override fun setListener() {
        ll_main.setOnClickListener(this)
        bt2.setOnClickListener(this)
        bt3.setOnClickListener(this)
    }

    override fun onClick(v: View?) {
        when (v?.id) {
            R.id.ll_main -> startAni()
            R.id.bt2 -> startAni2()
            R.id.bt3 -> startAni3()
            else -> {
            }
        }
        super.onClick(v)
    }

    private fun startAni() {
        ViewHelper.setY(bt1, 100f)
    }

    private fun startAni2() {
        //        ViewHelper.setY(bt1,100);
        ViewHelper.setTranslationY(bt1, 100f)
    }

    private fun startAni3() {
        ViewHelper.setScrollX(ll_main, 100)
        ViewHelper.setScrollX(bt3, -100)
    }
}
